### PR TITLE
Adding Ember documentation

### PR DESCRIPTION
**Pull request by nab880**
Adding ember documentation and examples

Erase this information and put your Pull Request comments here 
---

Instructions for Issuing a Pull Request to sst-elements
-------------------------------------------------------

1 - Verify that the Pull Request is targeted to the **devel** branch of sstsimulator/sst-elements

2 - Verify that Source branch is up to date with the devel branch of sst-elements

3 - After submitting your Pull Request:
   * Automatic Testing will commence in a short while 
      * Pull Requests will be tested with the devel branches of the sst-core and sst-sqe repositories
         * These branches are syncronized with the devel branch of sst-elements.  This is why is it important to keep your source branch up to date.
      * If testing passes, the source branch will be automatically merged (if possible)
         * Pull Requests from forks will not be automatically tested until the code is inspected.
         * Pull Requests from forks will not be automatically merged into the devel branch.
      * If testing fails, You will be notified of the test results.  
         * The Pull Request will be retested on a regular basis - Changes to the source branch can be made to correct problems
         
4 - DO NOT DELETE THE BRANCH (OR FORKED REPO) UNTIL THE PULL REQUEST IS MERGED.
----

### CODE REVIEW

File: src/sst/elements/ember/Makefile.am [hunk 1]
```diff
@@ -12,6 +12,8 @@ compdir = $(pkglibdir)
 comp_LTLIBRARIES = libember.la
 
 libember_la_SOURCES = \
+	example/motifs/emberexample.h
```
Comment (jpkenny) [inline src/sst/elements/ember/Makefile.am:15]: need the "\\" after these

**Review by jpkenny — CHANGES_REQUESTED** [review]
Just fix the makefile problem and I'll release the pre-check requirement

File: src/sst/elements/ember/Makefile.am (hunk 1 above)
Comment (jpkenny) [reply src/sst/elements/ember/Makefile.am]: fixed with changeset [f5a4ac1](https://github.com/sstsimulator/sst-elements/pull/1948/commits/f5a4ac14692fcad873abe6a944766bceee3d4565)

**Review by jpkenny — APPROVED** [review]
LGTM

File: src/sst/elements/ember/example/motifs/emberexample.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#ifndef _H_EMBER_FOO
+#define _H_EMBER_FOO
+
+#include "../../mpi/embermpigen.h"
+
+namespace SST {
+namespace Ember {
+
+class Foo : public EmberMessagePassingGenerator {
```
Comment (jpkenny) [inline src/sst/elements/ember/example/motifs/emberexample.h:24]: Foo -> Example
@nab880

**Comment by hughes-c** [comment]
```
In file included from example/motifs/emberexample.cc:18:
example/motifs/emberexample.h:29:9: error: expected identifier before string constant
         "ember",
         ^~~~~~~
example/motifs/emberexample.h:29:9: error: expected ',' or '...' before string constant
example/motifs/emberexample.h:34:5: error: ISO C++ forbids declaration of 'SST_ELI_REGISTER_SUBCOMPONENT_DERIVED' with no type [-fpermissive]
     );
```

**Comment by feldergast** [comment]
Adding WIP to this for now to avoid blocking core PRs needed for the release

**Comment by feldergast** [comment]
Converting to draft to make sure this doesn't accidentally merge.  The PR has over 700 commits in it, but only 10 changed files.  If you intend to merge it, please do a rebase against current devel.

**Pull request closed without merging**